multiregionccl: gate ADD REGION under ccl / enterprise license

otan opened the pull request:
Release note (enterprise change): ALTER DATABASE ... ADD REGION ... now
requires an enterprise license.

Release justification: low risk fix for new functionality

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/61169)
<!-- Reviewable:end -->

otan committed:
multiregionccl: gate ADD REGION under ccl / enterprise license

Release note (enterprise change): ALTER DATABASE ... ADD REGION ... now
requires an enterprise license.

Release justification: low risk fix for new functionality

ajstorm reviewed (approved):
Keep the butter coming!

> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/61169#-:-MUUqkG23A7lrH2UO9oo:b-1godp7)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
bors r=ajstorm
